File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n=nums.size();
        unordered_map<int,int> mp;
        for(int i=0;i<n;i++){
        int rem=target-nums[i];
        if(mp.find(rem)!=mp.end()){
            return {mp[rem],i};
        }
        mp[nums[i]]=i;
        }
        return {};
    }
};

File: 1-two-sum/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        seen={}
        for i,x in enumerate(nums):
            complement=target-x
            if complement in seen:
                return (seen[complement],i)
            seen[x]=i
        